src/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedCmsTests.netcoreapp.cs

```diff
@@ -0,0 +1,498 @@
+                    counterSigner.IncludeOption = X509IncludeOption.EndCertOnly;
+                    counterSigner.DigestAlgorithm = new Oid(Oids.Sha1, Oids.Sha1);
+                }
+
+                firstSignerInfo.ComputeCounterSignature(counterSigner);
+                Assert.True(signedHashCalledForCounterSignerKey(counterSignerKey));
+            }
+
+            byte[] counterSigned = cms.Encode();
+            Assert.True(ByteArrayContains(counterSigned, content));
+            Assert.True(counterSigned.Length > signed.Length);
+
+            return (counterSigned, firstSignerCert, counterSignerCert);
+        }
+
+        private static bool ByteArrayContains(byte[] array, byte[] subArray)
+        {
+            int lengthDiff = array.Length - subArray.Length;
+
+            if (lengthDiff < 0)
+            {
+                return false;
+            }
+
+            for (int shift = 0; shift <= lengthDiff; shift++)
+            {
+                if (subArray.SequenceEqual(array.Skip(shift).Take(subArray.Length)))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static (X509Certificate2 pubCert, AsymmetricAlgorithm key) GetRSACertAndKey()
+        {
+            using (X509Certificate2 signerCert = Certificates.RSA2048SignatureOnly.TryGetCertificateWithPrivateKey())
+            {
+                Assert.True(signerCert.HasPrivateKey);
+                X509Certificate2 pubCert = new X509Certificate2(signerCert.Export(X509ContentType.Cert));
+                Assert.False(pubCert.HasPrivateKey);
+
+                MyRSA key = new MyRSA(signerCert.GetRSAPrivateKey());
+                return (pubCert, key);
+            }
+        }
+
+        private static (X509Certificate2 pubCert, AsymmetricAlgorithm key) GetDSACertAndKey()
+        {
+            using (X509Certificate2 signerCert = Certificates.Dsa1024.TryGetCertificateWithPrivateKey())
+            {
+                Assert.True(signerCert.HasPrivateKey);
+                X509Certificate2 pubCert = new X509Certificate2(signerCert.Export(X509ContentType.Cert));
+                Assert.False(pubCert.HasPrivateKey);
+
+                MyDSA key = new MyDSA(signerCert.GetDSAPrivateKey());
+                return (pubCert, key);
+            }
+        }
+
+        private static (X509Certificate2 pubCert, AsymmetricAlgorithm key) GetECDsaCertAndKey()
+        {
+            using (X509Certificate2 signerCert = Certificates.ECDsaP256Win.TryGetCertificateWithPrivateKey())
+            {
+                Assert.True(signerCert.HasPrivateKey);
+                X509Certificate2 pubCert = new X509Certificate2(signerCert.Export(X509ContentType.Cert));
+                Assert.False(pubCert.HasPrivateKey);
+
+                MyECDsa key = new MyECDsa(signerCert.GetECDsaPrivateKey());
+                return (pubCert, key);
+            }
+        }
+
+        class MyRSA : RSA
```

Comment:
I'm not sure these classes belong.  If there's a justification for them that I'm not seeing then maybe we can avoid the "My" prefix?  (In X509Certificates the 3rd party RSA type is "RSAOther")

Reply:
removing them, we can't sign without private key provided externally anyway